### PR TITLE
Issue 3 user can view training log

### PR DESCRIPTION
**Pull request by ryan-byrnes**
https://user-images.githubusercontent.com/20606439/133800295-09026917-773d-49cd-8419-4709eedf8849.mov

### CODE REVIEW

File: client/components/training-session.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const response = await fetch(`/api/training/${sessionDate}`);
+    const trainingSession = await response.json();
+    this.setState({ trainingSession });
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props.date !== prevProps.date) {
+      await this.setState({
+        startDate: this.props.date
+      });
```
Comment (thebearingedge) [inline client/components/training-session.jsx:25]: This doesn't need to be `async/await`.

File: client/components/training-session.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          this.setState({ trainingSession });
+        });
+    }
+  }
+
+  render() {
+    const session = this.state.trainingSession;
+    if (!session) {
+      return null;
+    }
```
Comment (thebearingedge) [inline client/components/training-session.jsx:40]: This guard shouldn't be necessary if your initial `trainingSession` is an array. It's safe to call `.map()` on an empty array, it just gives you another empty array.

File: client/components/training-session.jsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    super(props);
+    this.state = {
+      trainingSession: [],
+      startDate: this.props.date
+    };
+  }
+
+  async componentDidMount() {
+    const date = moment(this.state.startDate).format();
+    const sessionDate = date.slice(0, 10);
```
Comment (thebearingedge) [inline client/components/training-session.jsx:15]: I don't feel great about this date formatting and slicing thing. Too much hidden intent.

I _think_ you are trying to "floor" the date?

Moment has a way for you to calculate that
https://momentjs.com/docs/#/manipulating/start-of/

And you can then format the string exactly as you want
https://momentjs.com/docs/#/displaying/format/

Using those will help the reader of your code a lot.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by ryan-byrnes** [comment]
https://user-images.githubusercontent.com/20606439/133907739-5e25f36c-048e-4a37-9fb7-4469abb860a5.mov

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**